lib/libc/newlib/libc-hooks.c

```diff
@@ -144,6 +148,12 @@ int _open(const char *name, int mode)
 }
 FUNC_ALIAS(_open, open, int);
 
+#else
+/* Just a prototype */
+int _write(int fd, char *buf, int nbytes);
+int _open(const char *name, int mode);
+#endif	/* CONFIG_COVERAGE_HW_TEST */
+
```

Comment:
are these functions used only in `coverage_dump()` or some instrumented gcc code calls them ?
If it is only used in `coverage_dump()` you don't need it, you can change that function to call any function you want. I don't like it that much, you are changing the behavior of these function when coverage is enabled.